src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private ActionStrategy strategy = new ActionStrategyImpl();
```

Comment:
You need to pass the ActionStrategy object as a parameter in the constructor, rather than initializing it here by yourself